### PR TITLE
mcu/nrf5340: Add RAM location check to calls using easyDMA

### PR DESCRIPTION
**Pull request by vikrant-proxy**
Add checks to confirm that the buffer points to a location in RAM given that easyDMA can only access RAM locations

### CODE REVIEW

**Review by vrahane — APPROVED** [review]
I am fine with the change, quite useful actually as we have seen. @kasjer The spiflash DFU issue was actually because of this. Let’s wait for others approval.

File: hw/mcu/nordic/nrf5340_net/src/hal_spi.c
Comment (kasjer) [inline hw/mcu/nordic/nrf5340_net/src/hal_spi.c:694]: This looks slightly strange to return SYS_EINVAL and EINVAL so close together for similar reason.
Maybe some unification could be applied, I'm not sure which way though.

**Review by kasjer — APPROVED** [review]
Even with those return error discrepancies it should be useful, I guess this kind of error happens only in development and is not really runtime error that will popup later.

File: hw/mcu/nordic/nrf5340_net/src/hal_spi.c
Comment (vikrant-proxy) [reply hw/mcu/nordic/nrf5340_net/src/hal_spi.c:694]: ah yes, let me be consistent with the existing convention.

Comment (vikrant-proxy) [reply hw/mcu/nordic/nrf5340_net/src/hal_spi.c]: addressed

**Pull request merged**
